explorer/client/src/utils/api/searchUtil.ts

```diff
@@ -1,104 +1,96 @@
+            console.error(err);
+            return null;
         });
-        searchPromises.push(addrObjPromise);
-    }
-
-    if (searchPromises.length === 0) {
-        navigate(`../error/all/${encodeURIComponent(input)}`);
-        return;
-    }
+};
 
-    return (
-        Promise.any(searchPromises)
-            .then((pac: any) => {
-                // Redirect to search result page if there are multiple categories with the same query
-                if (Array.isArray(pac)) {
-                    navigate(`../search-result/${encodeURIComponent(input)}`);
-                    return;
-                }
+const getDataOnObject = async (input: string, network: Network | string) => {
+    if (!isValidSuiAddress(input) && !isGenesisLibAddress(input)) return null;
+    return rpc(network)
+        .getObject(input)
+        .then((data) => {
+            if (data.status !== 'Exists') {
+                throw new Error('no object found');
+            }
+            return {
+                input: input,
+                category: 'objects',
+                result: data,
+            };
+        })
+        .catch((err) => {
+            console.error(err);
+            return null;
+        });
+};
 
-                if (
-                    pac?.data &&
-                    (pac?.category === 'objects' ||
-                        pac?.category === 'addresses' ||
-                        pac?.category === 'transactions')
-                ) {
-                    navigate(
-                        `../${pac.category}/${encodeURIComponent(input)}`,
-                        {
-                            state: pac.data,
-                        }
-                    );
-                } else {
-                    throw new Error(
-                        'Something wrong with navigateWithUnknown function'
-                    );
-                }
-            })
-            //if none of the queries find a result, show missing page
-            .catch((error) => {
-                // encode url in
-                navigate(`../error/missing/${encodeURIComponent(input)}`);
-            })
-    );
+export const navigateWithCategory = async (
+    input: string,
+    category: string,
+    network: string
+): Promise<{
+    input: string;
+    category: string;
+    result: SuiTransactionResponse | GetObjectDataResponse | SuiObjectInfo[];
+} | null> => {
+    switch (category) {
+        case 'transaction':
+            return getDataOnTx(input, network);
+        case 'object':
+            return getDataOnObject(input, network);
+        case 'address':
+            return getDataOnAddress(input, network);
+        case 'owner':
+            // The owner could be an object or an address
+            // first check for an object...
+            const objResult = await getDataOnObject(input, network);
+            // if no object check for an address
+            return objResult ? objResult : getDataOnAddress(input, network);
```

Comment:
Not clear how you handle instances where the ID is an address and Object or some variation of that?

Reply:
This is an excellent point to raise. I've updated the logic for this; let me know if there are any queries.